File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ComboBox/PopulatingData/CollectionDataSourcePage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Forms.CompanionKit.Controls.Util;

#endregion

namespace CompanionKit.Controls.ComboBox.PopulatingData
{
    public partial class CollectionDataSourcePage : UserControl
    {
        public CollectionDataSourcePage()
        {
            InitializeComponent();
            //Set collection of customer as DataSource.
            this.mobjComboBox.DataSource = CustomerData.GetCustomers();
        }

        /// <summary>
        /// Handles the SelectedIndexChanged event of the mobjComboBox control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            //Update text of represented TextBoxes with Value and Name of selected item.
            if (this.mobjComboBox.SelectedValue != null)
            {
                this.mobjValueTextBox.Text = this.mobjComboBox.SelectedValue.ToString();
            }
            this.mobjNameTextBox.Text = this.mobjComboBox.Text;
        }

        /// <summary>
        /// Handles the TextChanged event of the mobjValueMemberTextBox control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjValueMemberTextBox_TextChanged(object sender, EventArgs e)
        {
            if (!string.IsNullOrEmpty(mobjValueMemberTextBox.Text))
            {
                try
                {
                    mobjComboBox.ValueMember = mobjValueMemberTextBox.Text;
                }
                catch (Exception ex) { mobjErrorProvider.SetError(mobjValueMemberTextBox, ex.Message); }
                finally { mobjValueMemberTextBox.Text = mobjComboBox.ValueMember; }
            }
        }

        /// <summary>
        /// Handles the TextChanged event of the mobjDataTextBox control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjDataTextBox_TextChanged(object sender, EventArgs e)
        {
            if (!string.IsNullOrEmpty(mobjDataTextBox.Text))
            {
                foreach (Gizmox.WebGUI.Forms.CompanionKit.Controls.Util.Customer mobjItem in mobjComboBox.Items)
                {
                    if (mobjItem.FullName == mobjDataTextBox.Text)
                    {
                        mobjComboBox.SelectedItem = mobjItem;
                    }
                }
            }
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ComboBox/PopulatingData/CollectionDataSourcePage.vb
Namespace CompanionKit.Controls.ComboBox.PopulatingData

	Public Class CollectionDataSourcePage

        Public Sub New()
            InitializeComponent()
            'Set collection of customer as DataSource.
            Me.mobjComboBox.DataSource = Utils.CustomerData.GetCustomers()
        End Sub

        ''' <summary>
        ''' Handles the SelectedIndexChanged event of the mobjComboBox control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjComboBox_SelectedIndexChanged(sender As Object, e As EventArgs) Handles mobjComboBox.SelectedIndexChanged
            'Update text of represented TextBoxes with Value and Name of selected item.
            If Me.mobjComboBox.SelectedValue IsNot Nothing Then
                Me.mobjValueTextBox.Text = Me.mobjComboBox.SelectedValue.ToString()
            End If
            Me.mobjNameTextBox.Text = Me.mobjComboBox.Text
        End Sub

        ''' <summary>
        ''' Handles the TextChanged event of the mobjValueMemberTextBox control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjValueMemberTextBox_TextChanged(sender As Object, e As EventArgs) Handles mobjValueMemberTextBox.TextChanged
            If Not String.IsNullOrEmpty(mobjValueMemberTextBox.Text) Then
                Try
                    mobjComboBox.ValueMember = mobjValueMemberTextBox.Text
                Catch ex As Exception
                    mobjErrorProvider.SetError(mobjValueMemberTextBox, ex.Message)
                Finally
                    mobjValueMemberTextBox.Text = mobjComboBox.ValueMember
                End Try
            End If
        End Sub

        ''' <summary>
        ''' Handles the TextChanged event of the mobjDataTextBox control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjDataTextBox_TextChanged(sender As Object, e As EventArgs) Handles mobjDataTextBox.TextChanged
            If Not String.IsNullOrEmpty(mobjDataTextBox.Text) Then
                For Each mobjItem As Utils.Customer In mobjComboBox.Items
                    If mobjItem.FullName = mobjDataTextBox.Text Then
                        mobjComboBox.SelectedItem = mobjItem
                    End If
                Next
            End If
        End Sub
	End Class

End Namespace